presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -110,9 +114,15 @@
     @GuardedBy("root")
     private final Set<QueryExecution> runningQueries = new HashSet<>();
     @GuardedBy("root")
+    private LinkedHashSet<QueryExecution> activeQueries = new LinkedHashSet<>();
```

Comment:
I thought about this more last night, and I'm not sure there's value in having another data structure that is basically just queued+running. Let's not add another data structure that can get out of sync/leak.